lib/dry/logic/predicate.rb

```diff
@@ -4,17 +4,17 @@ def self.Predicate(block)
       case block
       when Method then Predicate.new(block.name, &block)
       when UnboundMethod then Predicate.new(block.name, fn: block)
-      else raise ArgumentError, 'predicate needs an :id'
+      else fail ArgumentError, 'predicate needs an :id'
       end
     end
 
     class Predicate
-      Undefined = Class.new {
+      Undefined = Class.new do
         def inspect
           "undefined"
         end
         alias_method :to_s, :inspect
-      }.new.freeze
+      end.new.freeze
```

Comment:
Please change it back to `{ .. }`, `do .. end` should be used in places where return value is not used, `end.new` is really ugly IMHO